### DISCUSSION TITLE
проверка по Npc_KnowsInfo // C_Npc_KnowsInfo

### DISCUSSION THREAD

**Discussion by Kor-Angar** [dosinabox/g2nr_unofficial_update#496] [General]
...

**Comment by Kor-Angar** [comment]
@dosinabox  напомни пожалуйста как работает это проверка без ЮНИОН ? если инстанции у  permanent  = 1 работает, а если нуль то не работает?

**Reply by dosinabox** [reply]
Диалоги с `permanent = 1` нельзя проверять с помощью `Npc_KnowsInfo`. Такие диалоги не будут считаться прослушанными.

**Reply by Kor-Angar** [reply]
становится с третьего раза чуть чуть понятнее.

**Comment by Kor-Angar** [comment]
проверку `Lares_Angekommen ` можно наверное заменить на проверку `C_Npc_KnowsInfo(DIA_Addon_Lares_ArrivedPortal);`

**Comment by Kor-Angar** [comment]
>> C_Npc_KnowsInfo

> Что-то новенькое?

это просто для себя в отладчик отправлял информацию поэтому  завернул внешнюю(?) функцию проверки Npc_KnowsInfo.

**Comment by Kor-Angar** [comment]
`C_Npc_KnowsInfo(DIA_Addon_Saturas_BeliarWeapGeben)` - ГГ отдал коготь белиара

подойдет для проверок в `DIA_NONE_100_Xardas` вместо  переменной `Saturas_KlaueInsMeer`

**Reply by dosinabox** [reply]
что есть C_Npc_KnowsInfo?

**Reply by Kor-Angar** [reply]
это `Npc_KnowsInfo(**hero**,DIA_Addon_Saturas_Lares)` с отсылкой инфы в отладчик.

**Comment by Kor-Angar** [comment]
надо будет проверить что возвращает `Npc_KnowsInfo`  без юнион для  `hero`. `other `и `self`

- [ ] Npc_KnowsInfo(**hero**,DIA_Addon_Saturas_Lares)
- [ ] Npc_KnowsInfo(**other**,DIA_Addon_Saturas_Lares)
- [ ] Npc_KnowsInfo(**self**,DIA_Addon_Saturas_Lares)

**Comment by Kor-Angar** [comment]
у Волка в диалоге про доспех.
```D
func int DIA_Wolf_BringPlates_Condition()
{
	if (Npc_KnowsInfo(hero,DIA_Wolf_AboutCrawler) && (Npc_HasItems(other,ItAt_CrawlerPlate) >= 10))
	//if((MIS_Wolf_BringCrawlerPlates == LOG_Running) && (Npc_HasItems(other,ItAt_CrawlerPlate) >= 10))
	{
		return TRUE;
	};
};
```

**Comment by Kor-Angar** [comment]
DIA_Karras_ItAm_Prot_BlackEye_Mis_Condition проверка по (Npc_KnowsInfo(hero,DIA_Karras_KarrasBlessedStone))

```D
func int DIA_Karras_ItAm_Prot_BlackEye_Mis_Condition()
{
	//if((hero.guild == GIL_KDF) && (MIS_Karras_FindBlessedStone == LOG_SUCCESS) && (DIA_Karras_ItAm_Prot_BlackEye_Mis_NoPerm == FALSE))
	if Npc_KnowsInfo(hero,DIA_Karras_KarrasBlessedStone)
	{
		return DIA_Karras_ItAm_Prot_BlackEye_Mis_NoPerm;
		//return TRUE;
	};
};
```

**Reply by Kor-Angar** [reply]
а	MIS_Karras_FindBlessedStone = LOG_SUCCESS лучше перенести 
из `DIA_Karras_KarrasBlessedStone_Info` в `DIA_Karras_ItAm_Prot_BlackEye_Mis_Info`

**Reply by Kor-Angar** [reply]
тогда можно  было бы 

```D
func int DIA_Karras_ItAm_Prot_BlackEye_Mis_Condition()
{
	//if((hero.guild == GIL_KDF) && (MIS_Karras_FindBlessedStone == LOG_SUCCESS) && (DIA_Karras_ItAm_Prot_BlackEye_Mis_NoPerm == FALSE))
	if (MIS_Karras_FindBlessedStone == LOG_SUCCESS)
	{
		return false;
	};
	if Npc_KnowsInfo(hero,DIA_Karras_KarrasBlessedStone)
	{
		return TRUE;
	};
};
```

**Comment by Kor-Angar** [comment]
у Ксардаса можно использовать проверку на 

	if Npc_KnowsInfo(hero,DIA_Addon_Saturas_BeliarWeapGeben)

вместо проверки
if (Saturas_KlaueInsMeer == TRUE)

**Comment by Kor-Angar** [comment]
вместо проверки  `if DJG_Angar_SentToStones == false` можно использовать проверку 
`if (!Npc_KnowsInfo(hero,DIA_GAROND_BACKINKAP4))`

**Reply by dosinabox** [reply]
Не соглашусь, там ещё проверяется DJG_AngarGotAmulett.

**Reply by Kor-Angar** [reply]
ты можешь конечно не согласится. но если герой уже отдал амулет Ангару то проверок по `DJG_Angar_SentToStones` уже не будет;-)
